### PR TITLE
Ouput registry information before prompting user for login information

### LINKED ISSUE #2071: [Feature] Print registry config value before on npm login
### Problem:
It is opaque which service will receive credentials for "npm login". This can lead to leaked credentials.

Logging into a non-standard registry is a 2-command process. Not only that, but one can be logged into multiple registries for various scopes. Give a multitude of registries, when I perform "npm login", depending on the flags, my credentials can be sent to a number of services. It would be nice to know exactly which one before credentials are entered.

### Solution:

Currently:
````
$ npm config set registry https://npm.fury.io/USERNAME/
$ npm login
Username: <user>
Password: <pass>
Email: (this IS public) <email>
Logged in as <user> on https://npm.fury.io/USERNAME/.
````

Proposed:
````
$ npm config set registry https://npm.fury.io/USERNAME/
$ npm login
Log in on https://npm.fury.io/USERNAME/
Username: <user>
Password: <pass>
Email: (this IS public) <email>
Logged in as <user> on https://npm.fury.io/USERNAME/.
````

### PR DESCRIPTION
**Pull request by Wicked7000**
This change outputs the registry before prompting the user for the login information.

Related to/Resolves: #2071

### CODE REVIEW

File: lib/auth/legacy.js [hunk 1]
```diff
@@ -57,6 +58,7 @@ const login = async (opts) => {
   }
 
   try {
+    output(`Log in on ${opts.registry}`)
```
Comment (ruyadorno) [inline lib/auth/legacy.js]: I believe it might be better to use `npmlog.notice` here instead 🤔

**Review by ruyadorno — COMMENTED** [review]
hi @Wicked7000 thanks for the awesome contribution!

Not a blocker to merge it since I believe we can address it later if the team has strong feelings about it but I believe this message would be better implemented as `npmlog.notice` for a few reasons:
- visual printed output is a little bit more highlighted
- that allow us to keep handling `output` only in `lib/adduser.js`
- provide users a way to opt-out of the message by tuning diff loglevels

File: lib/auth/legacy.js (hunk 1 above)
Comment (Wicked7000) [reply lib/auth/legacy.js]: Yeah I have no complaints with changing this out I just assumed we wouldn't want the extra text the notice adds before it, I'll change this up ASAP!

File: lib/auth/legacy.js [hunk 2]
```diff
@@ -57,6 +57,7 @@ const login = async (opts) => {
   }
 
   try {
+    log.notice('', `Log in on ${opts.registry}`)
```
Comment (ruyadorno) [inline lib/auth/legacy.js]: One more thing, we need to sanitize `opts.registry` output here using [`lib/utils/replace-info.js`](https://github.com/npm/cli/blob/latest/lib/utils/replace-info.js), e.g:

```diff
-    log.notice('', `Log in on ${opts.registry}`)
+    log.notice('', `Log in on ${replaceInfo(opts.registry)}`)
```

**Review by ruyadorno — APPROVED** [review]

**Pull request closed without merging**